steps/source/repo.py: Use --force-sync for sync step

willnewton opened the pull request:
Pass the --force-sync option to repo sync. This ensures that if
we change a module location in the manifest then the sync will
still succeed.

This option is not the default behaviour as it can cause data loss
in the working directory but this is not a concern for buildbot
as no changes are ever made in the checked out copy.

## Remove this paragraph
Please have a look at our developer documentation before submitting your Pull Request.

http://trac.buildbot.net/wiki/Development
And especially:
http://trac.buildbot.net/wiki/SubmittingPatches

## Contributor Checklist:

* [ ] I have updated the unit tests
* [ ] I have created a file in the `master/buildbot/newsfragment` directory (and read the `README.txt` in that directory)
* [ ] I have updated the appropriate documentation

tardyp commented:
change is okay, but you really need to do your usual homework...

- update the tests
- add newsframent
- update documentation (add a warning in the repo section)

willnewton commented:
@tardyp sorry, I fixed the tests and added the news fragment. I didn't add anything to the docs as I was unsure how to word it. From a user's perspective all this should do is make syncs that failed previously now succeed, I can't imagine a scenario where a user would have modifications in their checked out copy (it would also be overwritten if they used the tarball option).

tardyp commented on master/buildbot/newsfragments/repo_force_sync.feature:
```diff
@@ -0,0 +1 @@
+The repo source step now syncs with the --force-sync flag which allows the sync to proceed when a source repo in the manifest has changed.
```
you need to double-back-quote ``--force-sync``

willnewton commented:
@tardyp I updated the new fragment following your suggestion.

vvscloud commented:
I feel this should be updated in document, as this feature will make repo step fail with old workers. as --force-sync flag is unrecognized by old versions of repo. Did we test the compatibility of this change with old workers?

tardyp commented:
Is there any reason to use older version of repo? repo is supposed to be automatically updated to the latest version.
--force-sync has been added in 2014..

https://gerrit.googlesource.com/git-repo/+/abaa7f312f1b6c8d11d7c757fe909900ce5788b5
